repos/c2sm/packages/flexpart-cosmo/package.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#
+from spack import *
+from distutils.dir_util import copy_tree
+from llnl.util.filesystem import working_dir
+
+
+class FlexpartCosmo(MakefilePackage):
+    """flexpart is a Lagrangian dispersion model"""
+
+    homepage = 'https://github.com/C2SM-RCM/flexpart'
+    git = 'ssh://git@github.com/C2SM-RCM/flexpart.git'
+
+    version('main', branch='testing')
+
+    depends_on('eccodes +fortran')
+    depends_on('netcdf-fortran')
+
+    depends_on('libtool@2.4.6', type='build')
+    depends_on('eccodes@2.25:', type=('build', 'link', 'run'))
```

Comment:
I just tried to install flexpart-cosmo with eccodes 2.23.0 and this worked out fine. Are you sure it technically needs >2.25 to function?

Reply:
Flexpart people usually use eccodes 2.19.0 or 2.25.0 and don't mind using a newer version. But is it worth it to find out exactly which version it works with? I can do that, but I thought it was not worth the effort.

As you pointed out correctly, the run-time dependency is not needed. I will remove it.

Another question related to that: If I specify, let's say `depends_on('eccodes@2.23:')`, should I then specify `^eccodes@2.25.0` for the system test, as this will be used by the majority of users?

Reply:
@dominichofer Another point that we should be aware of: According to @pirmink, we also have to take the MeteoSwiss GRIB definitions into account, which have to be compatible with a certain eccodes version. On balfrin, there is `eccodes 2.25.0` and `eccodes_cosmo_resources/2.25.0.2`.

The `eccodes_cosmo_resources` should be compatible with newer versions of `eccodes` as long as no encodings are used that are not yet present in the `eccodes_cosmo_resources`.

There is also a spack package [cosmo-eccodes-definitions](https://github.com/C2SM/spack-c2sm/blob/flexpart/repos/c2sm/packages/cosmo-eccodes-definitions/package.py#L35). Packages like `int2lm`, `cosmo` and `icon` depend on `cosmo-eccodes-definitions`. Is this also necessary for flexpart? In that case I think `eccodes@2.25:` will indeed be a requirement.

Reply:
Spack is general purpose package manager, its responsibility is to resolve dependencies and provide the package with its requirements. I understand that flexpart people are fine with 2.25.0, but the restriction of `eccodes@2.25:` only stems from spack, not the flexpart repo, but an artificial restriction from spack. In my opinion that's invasive of spack and not nice. But it's okay for me if you put `depends_on('eccodes@2.25:')` in this PR. It's better to have a restricted cosmo-flexpart, than none. But it would be even better if there was no artificial restriction imposed by spack. You see what I mean?
We don't have to figure out which versions are compatible, we can just copy the requirements and restriction the flexpart repository specifies (in their readme); which doesn't say anything about eccodes' version I think. Or ask the flexpart devs. It is not a spack-dev's job to do dependency management. That's the flexpart-dev's job. We just copy and follow suit. Spack is a package manager, not a dependency-version-compatibility-tracker for packages.
This is just my personal opinion. If you want to stick with `depends_on('eccodes@2.25:')`, feel free to do so.

Reply:
> If I specify, let's say depends_on('eccodes@2.23:'), should I then specify ^eccodes@2.25.0 for the system test, as this will be used by the majority of users?

In my opinion, no. Because the test tries to verify that flexpart-cosmo/package.py is correct. Testing that flexpart-cosmo is compatible with eccodes is not the responsibility of spack-c2sm. Just a package manager, not a software tester. Therefore any version of eccodes is good enough for the test. And if you don't specify the version of eccodes, we will always get the latest one if new ones arrive. Doesn't hurt to try go with the flow and test the novel stuff.
This is just my opinion on the role of spack and I know others see this differently.